qiskit_nature/circuit/library/initial_states/fermionic_gaussian_state.py

```diff
@@ -0,0 +1,171 @@
+
+from .utils.givens_rotations import _prepare_fermionic_gaussian_state_jordan_wigner
+
+
+def _validate_transformation_matrix(
+    mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8
+) -> None:
+    if not len(mat.shape) == 2:
+        raise ValueError(
+            "transformation_matrix must be a 2-dimensional array. "
+            f"Instead, got shape {mat.shape}."
+        )
+
+    n, p = mat.shape  # pylint: disable=invalid-name
+    if p != n * 2:
+        raise ValueError(
+            "transformation_matrix must have shape (n_orbitals, 2 * n_orbitals). "
+            f"Instead, got shape {mat.shape}."
+        )
+
+    left = mat[:, :n]
+    right = mat[:, n:]
+    comm1 = left @ left.T.conj() + right @ right.T.conj()
+    comm2 = left @ right.T + right @ left.T
+    one = np.eye(n)
+    zero = np.zeros((n, n))
+    if not np.allclose(comm1, one, rtol=rtol, atol=atol) or not np.allclose(comm2, zero):
+        raise ValueError(
+            "transformation_matrix does not describe a valid transformation "
+            "of fermionic ladder operators. A valid matrix should have the block form "
+            "[W1 W2] where W1 @ W1.T.conj() + W2 @ W2.T.conj() = I and "
+            "W1 @ W2.T + W2 @ W1.T = 0."
+        )
+
+
+class FermionicGaussianState(QuantumCircuit):
+    r"""A circuit that prepares a fermionic Gaussian state.
+
+    A fermionic Gaussian state is a state of the form
+
+    .. math::
+        b^\dagger_1 \cdots b^\dagger_{N_p} \lvert \overline{\text{vac}} \rangle
+
+    where
+
+    .. math::
+        \begin{pmatrix}
+            b^\dagger_1 \\
+            \vdots \\
+            b^\dagger_N \\
+        \end{pmatrix}
+        = W
+        \begin{pmatrix}
+            a^\dagger_1 \\
+            \vdots \\
+            a^\dagger_N \\
+            a_1 \\
+            \vdots \\
+            a_N
+        \end{pmatrix},
+
+    - :math:`a^\dagger_1, \ldots, a^\dagger_{N}` are the fermionic creation operators
+    - :math:`W` is an :math:`N \times 2N` matrix such that
+      :math:`b^\dagger_1, \ldots, b^\dagger_{N}` also satisfy the
+      fermionic anticommutation relations
+    - :math:`\lvert \overline{\text{vac}} \rangle` is the mutual 0-eigenvector of
+      the operators :math:`\{b_j^\dagger b_j\}`
+
+    The matrix :math:`W` has the block form
+
+    .. math::
+        \begin{pmatrix}
+            W_1 & W_2
+        \end{pmatrix}
+
+    where :math:`W_1` and :math:`W_2` must satisfy
+
+    .. math::
+        W_1 W_1^\dagger + W_2 W_2^\dagger = I \\
+        W_1 W_2^T + W_2 W_1^T = 0
+
+    The matrix :math:`W` is commonly obtained by calling the
+    :meth:`~.QuadraticHamiltonian.diagonalizing_bogoliubov_transform`
+    method of the :class:`~.QuadraticHamiltonian` class.
+    This matrix is used to create circuits that prepare eigenstates of the
+    quadratic Hamiltonian.
+
+    Currently, only the Jordan-Wigner Transformation is supported.
+
+    Reference: arXiv:1711.05395
```

Comment:
It might be nice if the reference were an actual link in the resulting documentation for the user